src/kernel/context.cpp

```diff
@@ -10,13 +10,16 @@
 
 #include <string>
 
-
 namespace kernel {
 Context::Context()
 {
-    std::string sha256_algo = SHA256AutoDetect();
-    LogPrintf("Using the '%s' SHA256 implementation\n", sha256_algo);
-    RandomInit();
+    static bool initialized{false};
+    if (!initialized) {
+        std::string sha256_algo = SHA256AutoDetect();
+        LogPrintf("Using the '%s' SHA256 implementation\n", sha256_algo);
```

Comment:
While this code is being touched, do we want to use more specific logging?  (e.g. `LogDebug`, `LogInfo`, etc.)

https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#logging

Reply:
Aye, pushed.